### PR TITLE
feat: [GALI-2125] add fields to data set creation

### PR DESCRIPTION
**Pull request by aws-tyler**
Issue #, if available:

Description of changes:

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: common/changes/@aws/workbench-core-datasets/atmikev-GALI-2125_AddFieldsToDataSetCreation_2022-12-21-00-43.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (bhudnell) [inline common/changes/@aws/workbench-core-datasets/atmikev-GALI-2125_AddFieldsToDataSetCreation_2022-12-21-00-43.json:1]: is this change file necessary still?

Comment (aws-tyler) [reply common/changes/@aws/workbench-core-datasets/atmikev-GALI-2125_AddFieldsToDataSetCreation_2022-12-21-00-43.json:1]: nice catch.

**Review by bhudnell — APPROVED** [review]

File: solutions/swb-app/src/dataSets/createDataSetRequestParser.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import { z } from 'zod';
+
+// eslint-disable-next-line @rushstack/typedef-var
+export const CreateDataSetRequestParser = z.object({
+  name: z.string(),
```
Comment (FernandoAranda) [inline solutions/swb-app/src/dataSets/createDataSetRequestParser.ts:9]: do we want strict in here?

File: solutions/swb-app/src/dataSets/createExternalEndpointRequestParser.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import { z } from 'zod';
+
+// eslint-disable-next-line @rushstack/typedef-var
+export const CreateExternalEndpointRequestParser = z.object({
```
Comment (FernandoAranda) [inline solutions/swb-app/src/dataSets/createExternalEndpointRequestParser.ts:8]: same as above

**Review by FernandoAranda — APPROVED** [review]

File: solutions/swb-app/src/dataSets/createDataSetRequestParser.ts (hunk 2 above)
Comment (aws-tyler) [reply solutions/swb-app/src/dataSets/createDataSetRequestParser.ts:9]: Good catch. Adding that in with the next PR

File: solutions/swb-app/src/dataSets/createExternalEndpointRequestParser.ts (hunk 3 above)
Comment (aws-tyler) [reply solutions/swb-app/src/dataSets/createExternalEndpointRequestParser.ts:8]: Good catch. Adding that in with the next PR

**Pull request merged**